src/plugins/proxy/src/plugin.cpp

```diff
@@ -299,15 +300,35 @@ ov::Any ov::proxy::Plugin::get_property(const std::string& name, const ov::AnyMa
     return get_core()->get_property(get_primary_device(device_id), name, {});
 }
 
+ov::SoPtr<ov::IRemoteContext> ov::proxy::Plugin::create_proxy_context(
+    const ov::SoPtr<ov::ICompiledModel>& compiled_model,
+    const ov::AnyMap& properties) const {
+    auto dev_name = get_device_name();
+    auto dev_idx = get_device_from_config(properties);
+    auto has_dev_idx = is_device_in_config(properties);
+    auto is_new_api = get_core()->is_new_api();
+    ov::SoPtr<ov::IRemoteContext> device_context;
+    ov::SoPtr<ov::IRemoteContext> remote_context;
+    try {
+        device_context = compiled_model->get_context();
+        if (!device_context._so)
```

Comment:
when can we hit this if clause?

Reply:
Hardware plugins return context without so, but in case if undert proxy we use auto batch or another meta plugins, we already will have so object inside